testing/integration/endpoint_security_test.go

```diff
@@ -58,6 +61,212 @@ var protectionTests = []struct {
 	},
 }
 
+func TestUpgradeAgentWithTamperProtectedEndpoint_DEB(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: Deb,
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		OS: []define.OS{
+			{
+				Type: define.Linux,
+			},
+		},
+	})
+	testTamperProtectedDebRpmUpgrades(t, info, "deb")
+}
+
+func TestUpgradeAgentWithTamperProtectedEndpoint_RPM(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: RPM,
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		OS: []define.OS{
+			{
+				Type:   define.Linux,
+				Distro: "rhel",
+			},
+		},
+	})
+	testTamperProtectedDebRpmUpgrades(t, info, "rpm")
+}
+
+func getEndpointVersion(t *testing.T) string {
+	cmd := exec.Command("sudo", "/opt/Elastic/Endpoint/elastic-endpoint", "version")
+	output, err := cmd.CombinedOutput()
+	require.NoError(t, err)
+	// version: 8.18.0-SNAPSHOT, compiled: Wed Feb 19 01:00:00 2025, branch: HEAD, commit: c450b50f91507c3166b072df8557f5efd871103a
+	return strings.Split(strings.Split(string(output), "version: ")[1], ",")[0]
```

Comment:
What about something like this (optionally you can check for `found` return value:
```suggestion
	endpointVersionFragment, _, _ := strings.Cut(string(output), ",")
	endpointVersion, _ := strings.CutPrefix(endpointVersionFragment, "version: ")
```